### DISCUSSION TITLE
Shared hyperparameters between mean function and kernel

### DISCUSSION THREAD

**Discussion by CR-Richardson** [cornellius-gp/gpytorch#2614] [Q&A]
I would like to optimize the hyperparameters of a multi-output GP with the following mean function and kernel: 

$m(t) = C e^{At} \mu$

$k_{jq}(t,t') = C_{j:} e^{At} \Sigma e^{At'} C_{q:}^{\top}$

The vector $\mu$ and matrix $\Sigma$ are known and the matrices $A, B, C$ are hyperparameters to be learnt.

The kernel above defines the covariance between elements $j$ and $q$ of the multi-dimensional output. The notation $C_{j:}$ denotes row j of the matrix $C$.

Does anyone have any suggestions for how to structure the classes defining the mean function and kernel when the parameters are shared? And would this affect the boiler plate code for the model class?

**Comment by gpleiss** [comment]
Your model can be written as $y = Ce^{At} \theta + \epsilon$, where $\epsilon$ is observation noise and $\theta$ is a latent variable with mean $\mu$ and covariance $\Sigma$. While this model is technically a GP (w.r.t. t), I don't think GPyTorch is the right package to use because you are essentially trying to infer the finite-dimensional latent variable $\theta$.
